### ISSUE TITLE
Setting separate username for imap accounts

### ISSUE THREAD

**Issue by yash-ahir** [foxcpp/maddy#325]
Hello, after reading and setting up the mailserver according to the initial configuration page, I noticed this

> Note the username is a e-mail address. This is required as username is used to authorize IMAP and SMTP access (unless you configure custom mappings, not described here).

and

> Note that IMAP accounts and credentials are managed separately yet usernames should match by default for things to work.

I wasn't able to find any documentation related to this, is this technically infeasible with the current state of the project (as in incomplete feature)? If not can you source some documentation where I can set this up? My email address is `mail@example.org` and as such the user's name appears as `mail`, which is really the only thing irking me out about this wonderful project. So is there any way to change that up?

**Comment by foxcpp** [comment]
This is a feature that will be available in 0.5, but default will remain as it is currently.
In fact it is already implemented on the `dev` branch: 
https://github.com/foxcpp/maddy/commit/a2e781ab3aaad7ca8300efb60ed032f2bd7a6114

In particular, to have usernames without domain part something like this can be used:
```
storage.imapsql local_mailboxes {
  ...
  auth_map regexp "(.*)" "$1@example.org"
  auth_normalize precis # default is precis_email and it will break if username is not a email
}
```

Note the comment in documentation:
>imapsql unconditionally requires a full email as an account name as they are
used to determine the right mailboxes for inbound messages. If you do not
want domain-specific accounts then you can use a dummy domain like
foxcpp@maddy.invalid. But you will need to ensure that on delivery recipient
addresses are rewritten appropriately (e.g.
foxcpp@example.org => foxcpp@maddy.invalid).

Another note if you want to use maddy version built from `dev` branch - this branch does not contain bugfixes from 0.4.x so you would need to do `git merge master` before building it. (I hope it merges cleanly... It should)

**Comment by yash-ahir** [comment]
I see, thank you for the information, marking this as closed.
